gns/train.py

```diff
@@ -178,11 +178,13 @@ def train(rank, flags, world_size, device):
   Args:
     rank: local rank
     world_size: total number of ranks
+    device: torch device type
   """
   if device == torch.device("cuda"):
     distribute.setup(rank, world_size, device)
 
   metadata = reading_utils.read_metadata(flags["data_path"])
+  metadata = metadata["train"]
```

Comment:
You are overwriting `metadata?` why not just concatenate and get the relevant data?